demo/simple-column-sorting-pipeline.html

```diff
@@ -0,0 +1,155 @@
+<!DOCTYPE html>
```

Comment:
so this is nice but, as I've mentioned, I believe the purpose of the ticket is to **provide a way to replace the comparators** — not necessarily the the entire sorting transformer.

_(I do understand the advantage you explained to me, that it no longer uses Steve Wirt's code. Not sure why that's a priority. Steve did some interesting things and it has a number of advantages, specifically that it handles multi-column sorts and does so relatively efficiently.)_

So to complete this ticket I believe we need to provide a way to specify your own comparators. I already gave you my thoughts on that:

1. When a comparator is given for a column, use it!
2. When a comparator is not given, but a type is, use that to pick a standard comparator.
3. When type is not known, infer number vs string from first row.

Item 3 above is the way it works now. Let's add 1 and 2.

I also pointed out the following observations about the way it chooses comparators now:

1. It uses separate comparators for number (a-b) and string (a < b, a > b). The latter would clearly work for both but I suppose the thinking was that the former is more efficient for numbers (one arithmetic operation instead of two)? Maybe we should profile or jsPerf it to see if it make a significant difference.
2. It uses separate ascending and descending comparators. This is necessary because Array#sort does not have an option for direction. So our app developers will need to supply comparators in pairs.

I also suggested that comparators be supplied in the same way we are currently supplying `fields` (column names) and `calculators`, which are discrete array parameters to DataSourceOrigin constructor. We could just add another parameter:


```javascript
new DataSourceOrigin(data, fields, calculators, comparators)
```

**_HOWEVER_** on further reflection, I think it is time we supplied a single array instead:

```javascript
new DataSourceOrigin(data, fields)
```
where _each element_ of `fields` would be defined as follows:
* string - column name
* object with following properties:
   * `name` _(required)_ column name
   * `calculator:` _(optional)_ a function for a computed column (only)
   * `comparator:` _(optional)_ object:
      * `asc:` ascending comparator
      * `desc:` descending comparator